File: backend/src/main/java/edu/cit/amad/travellite/dto/AuthResponse.java
package edu.cit.amad.travellite.dto;

public class AuthResponse {

    private boolean success;
    private Object data;
    private Object error;
    private String timestamp;

    public AuthResponse(boolean success, Object data, Object error, String timestamp) {
        this.success = success;
        this.data = data;
        this.error = error;
        this.timestamp = timestamp;
    }

    public boolean isSuccess() { return success; }
    public Object getData() { return data; }
    public Object getError() { return error; }
    public String getTimestamp() { return timestamp; }

    public void setSuccess(boolean success) { this.success = success; }
    public void setData(Object data) { this.data = data; }
    public void setError(Object error) { this.error = error; }
    public void setTimestamp(String timestamp) { this.timestamp = timestamp; }
}

File: mobile/app/src/main/java/edu/cit/amad/travellite/model/AuthResponse.kt
package edu.cit.amad.travellite.model

data class AuthResponse(
    val success: Boolean,
    val data: AuthData?,
    val error: ErrorData?,
    val timestamp: String?
)

data class AuthData(
    val user: UserData?,
    val accessToken: String?,
    val refreshToken: String?
)

data class UserData(
    val first_name: String?,
    val last_name: String?,
    val email: String?
)

data class ErrorData(
    val code: String?,
    val message: String?,
    val details: Any?
)